lib/src/views/Calendar/YearSelection.tsx

```diff
@@ -70,6 +73,21 @@ export const YearSelection: React.FC<YearSelectionProps> = ({
     (year: number, isFinish = true) => {
       const newDate = utils.setYear(selectedDate, year);
       if (isDateDisabled(newDate)) {
+        const closestEnabledDate = findClosestEnabledDate({
+          utils,
+          date: newDate,
+          minDate,
+          maxDate,
+          disablePast: Boolean(disablePast),
+          disableFuture: Boolean(disableFuture),
+          shouldDisableDate: isDateDisabled,
+        });
+
+        onChange(closestEnabledDate, isFinish);
+        if (onYearChange) {
+          onYearChange(newDate);
+        }
```

Comment:
The call order between onChange and onYearChange isn't stable. Should be we care about it?

Reply:
Fixed